Automattic/wp-calypso
# Theme Showcase: Implement Theme Preview modal component

#### Proposed Changes

This PR implements a new theme preview modal for the Theme Showcase, building on top of the `@automattic/design-preview` components.

![Screen Shot 2023-01-03 at 11 18 18 AM](https://user-images.githubusercontent.com/797888/210296064-5f2f8601-00cc-4bb2-bc0b-e59b3a20be72.png)

The scope of this PR is only the theme preview modal in isolation, which means that the end-to-end interaction flow is not implemented here. Specifically: 

1. Carrying over selected style variation from the theme card to the theme preview.
2. Post activation flow in the theme preview (it's kept the same as the one in production).
3. The selected style variation will not be applied to the activated theme.

#### Testing Instructions



* Head to the logged-in Theme Showcase `/themes/${site_slug}`. If using calypso.live, the flag `themes/showcase-i4/details-and-preview` is required. 
* Select any theme except the active theme, and click on the "Open live demo" button in the theme details page.
* Ensure that the new preview modal shows up, and is as per design spec in 0EogtxvlmxMGcnEEaMVECv-fi-1401%3A53070&t=w0yGbxu7F2y0vApp-0.
* Ensure that the new preview modal also works in the logged-out Theme Showcase (https://wordpress.com/themes).
* Also test the new preview modal in an Atomic site.

#### Pre-merge Checklist
